### PR TITLE
random rec change, warnings, association rules refactoring

### PR DESCRIPTION
**Pull request by monkey0head**
random rec changed: 
- random generator as a new attribute to provide recommendations diversity and reproducibility for model with fixed seed 
- fit for uniform model distinguished 
- probabilities for uniform predict are not passed to udf (speed up)
- np.random.choice is replaced with default_rng().choice() (new way of sampling)
- tests added

warnings for neural models and wrappers added
association rules code refactoring
dataframe non-equality check added to utils

### CODE REVIEW

File: tests/models/test_random_rec.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+def test_works(log, uniform_seed, uniform_no_seed, popular_based_seed):
+    for model in [uniform_seed, uniform_no_seed, popular_based_seed]:
+        model.fit(log)
+        model.item_popularity.count()
+        model.item_popularity.show()
+
+
+def test_popularity_matrix(
+    log, uniform_seed, uniform_no_seed, popular_based_seed
```
[the remark this answers is not in the document]
Comment (monkey0head) [reply tests/models/test_random_rec.py]: Thanks, you are right. I did it a little differently using `fitted_model` fixture with `params` to run tests for all fitted models without cycles

**Review by Darel13712 — APPROVED** [review]

**Comment by monkey0head** [comment]
@AleXXL1986 please have a look

**Review by AleXXL1986 — APPROVED** [review]
